Stn 653  spotlight color dropdown

lunarfusion opened the pull request:
# READY FOR REVIEW

## Summary
Added option on the Spotlight paragraph type to select a global gray background color (#D9D7D2) or the 
 theme's default Spotlight background color.

View of select list options:
<img width="755" alt="Screen Shot 2022-02-16 at 9 16 26 AM" src="https://user-images.githubusercontent.com/25614178/154287682-79ae55ca-e763-4f98-a534-d917558f7e41.png">

View of results in Colorful/Cardinal:
<img width="1214" alt="Screen Shot 2022-02-16 at 9 55 02 AM" src="https://user-images.githubusercontent.com/25614178/154291256-d67a1461-ced0-419f-846e-79a50fd3c615.png">

View of results in Colorful/Mountain:
<img width="1217" alt="Screen Shot 2022-02-16 at 9 55 21 AM" src="https://user-images.githubusercontent.com/25614178/154291314-ede643a6-64dc-435f-bf17-6bd79c0bfdee.png">

Jira Card: https://sparkbox.atlassian.net/jira/software/c/projects/STN/boards/80?modal=detail&selectedIssue=STN-653

## Need Review By (Date)
asap

## Urgency
medium

## Steps to Test
1. npm run test
2. Create a new page and add a Spotlight paragraph type to that page.
3. Under the "Background Color" select list, select "Default background" and verify that the theme's default spotlight background appears.
4. Under the "Background Color" select list, select "Medium Gray background" and verify that the global medium gray background #D9D7D2 appears.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)

lunarfusion committed:
feat(STN-653) spotlight color dropdown

Spotlight background gray option added

lunarfusion committed:
chore: spotlight background colors config

rkelbel reviewed (commented):
Monet, first of all, this is awesome work! Congrats on the first SBX PR request! 🎉 💥 😄 

This is work is very close but I did have some things for feedback that we will want to address or discuss before merging this work.

rkelbel reviewed (changes requested):
There seems to be some configuration missing from this work that fully adds the configuration for the template to add the class name from the new field consistently.

Namely, When I create a Short Spotlight slide, the class is not populating for me on my spotlight slide.  It is however populating correctly for the Tall spotlight.

This is due to the configuration between "Default" (Tall) and "Preview" (Short) Displays not being the same in their field display for the background select list field.

Preview (Short):
<img width="965" alt="image" src="https://user-images.githubusercontent.com/9355698/154779870-f8def99b-aa21-4015-8b8b-2b832fc723bd.png">

Default (Tall):
<img width="931" alt="image" src="https://user-images.githubusercontent.com/9355698/154779902-bd3ffe35-1fae-4c8e-a632-7e2a6d0cc333.png">

You can see between those two screenshots that the field is not in the Background area on short and it is on tall. This likely happened because you and I missed the Preview configuration display file when we paired to export the configuration.
If you do another export you can find this missing file and export it by itself and discard any of the other config that spits out. The file you will be looking for is: `core.entity_view_display.paragraph.hs_spotlight.preview`.
Once you export that file, we should be able to import the short display variant and see the class properly populating.

rkelbel reviewed (changes requested):
Some additional config changes we will want to make, is to make the Label "hidden" for the field on both the Preview and Default displays for Short and Tall. Although this field is not actually showing up, I think we should cover our bases by removing a possible label from the equation.

<img width="763" alt="image" src="https://user-images.githubusercontent.com/9355698/154780394-5b9ce68a-e683-4887-9092-35457c69bac8.png">

Additionally, I believe the request from Tori was for the list options to reflect similar listings as Color Band.  Currently, we have the background gray option listed as "Medium Gray background" in the select list.  
I think this could be simplified to just say "Gray" as there is not as of now an additional gray selection. 
We can also simplify "Default background" to "Default". Similar to Color Bands options: https://sparkbox-sandbox.suhumsci.loc/admin/structure/paragraphs_type/hs_clr_bnd/fields/paragraph.hs_clr_bnd.field_hs_clr_bnd_bgc/storage

<img width="248" alt="image" src="https://user-images.githubusercontent.com/9355698/154780368-7df981be-a166-4b3b-a152-459044122f75.png">

If you would like to verify this with Tori prior to updating this in configuration please do so, I just thought I would include this feedback for consideration and consistency.

rkelbel reviewed (changes requested):
The last config change we should make is also something H&S seems to prefer on fields like this.
We may want to make this field "Required" for the background color choice, setting the field to required removes "None" from the select list as an option. Although we have a Default color set, that would ensure "None" cannot be selected.

Although I do not see this as an Acceptance Criteria from H&S on the card, it is probably worth confirming before committing all of the configurations for this work.

<img width="204" alt="image" src="https://user-images.githubusercontent.com/9355698/154780625-8704dfab-3166-46c6-8764-7c52a9aa81ab.png">

rkelbel commented on config/default/core.entity_form_display.paragraph.hs_spotlight.default.yml:
```diff
@@ -3,6 +3,7 @@ langcode: en
 status: true
 dependencies:
   config:
+    - field.field.paragraph.hs_spotlight.field_hs_spotlight_bg
```
Perfect Field machine-id naming! 👍🏻

rkelbel commented on docroot/themes/humsci/humsci_basic/src/scss/components/_pattern.spotlight.scss:
```diff
@@ -2,6 +2,10 @@
+  &--background-gray {
+    @include hb-global-color('background-color', 'gray-medium');
+  }
+
```
Nice SASS! 💥

lunarfusion committed:
chore: adding additional spotlight configuration

changed field option names to gray and default, changed background field to required, hide background field label

lunarfusion committed:
feat(STN-653): update ocean secondary color to accessible

lunarfusion committed:
chore: corrected config with field in correct location and hidden label

lunarfusion committed:
chore: update config for short spotlight spotlight

lunarfusion committed:
chore: update default choice on background field config

Commit:
chore: compile css

rkelbel reviewed (approved):
This looks awesome @lunarfusion ! Nice work! 🎉 
![high five](https://media.giphy.com/media/1DfZvKmiELvtS/giphy.gif)
